support-lib/cpp/Future.hpp

```diff
@@ -16,6 +16,8 @@
 
 #pragma once
 
+#include "expected.hpp"
+
 #include <atomic>
 #include <functional>
 #include <memory>
```

Comment:
I find I have to change the feature detection to the following to make it work in c++17 + -fcoroutine-ts
```
#if defined(__cpp_coroutines) && __has_include(<experimental/coroutine>)
    #include <experimental/coroutine>
    namespace djinni::detail {
        template <typename Promise = void> using CoroutineHandle = std::experimental::coroutine_handle<Promise>;
        using SuspendNever = std::experimental::suspend_never;
    }
    #define DJINNI_FUTURE_HAS_COROUTINE_SUPPORT 1
#elif defined(__cpp_impl_coroutine) && __has_include(<coroutine>)
    #include <coroutine>
    namespace djinni::detail {
        template <typename Promise = void> using CoroutineHandle = std::coroutine_handle<Promise>;
        using SuspendNever = std::suspend_never;
    }
    #define DJINNI_FUTURE_HAS_COROUTINE_SUPPORT 1
#endif
```

Reply:
Looks like with these settings the `<coroutine>` header is available but not functional. Your suggestion fixes that nicely, but I found another issue: with clang 16 and `-std=c++20 -stdlib=libc++`, it chooses the `experimental` implementation even though the proper one is available.

I've done some [experiments](https://godbolt.org/z/1enjqz9E3):
- `__cpp_coroutines` is available for both C++20 and C++17 + `-fcoroutines-ts` until clang 17 when the macro and `-fcoroutines-ts` were completely removed
- `__cpp_impl_coroutine` and `__cpp_lib_coroutine` are only available in proper C++20 mode when we have `<coroutine>`

I've just pushed another suggestion based on these findings. Here's what I used to test it with a few compilers/settings: https://godbolt.org/z/rj737E5eq

Caveat: I've tested only with clang. For iOS and Android that seems to be the relevant compiler.

Reply:
thanks! this works really well.